pkg/scheduler/framework/plugins/clustereligibility/plugin.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package clustereligibility features a scheduler plugin that filters out clusters
+// that are not eligible for resource placement.
+package clustereligibility
+
+import (
+	"context"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler/clustereligibilitychecker"
+	"go.goms.io/fleet/pkg/scheduler/framework"
+)
+
+const (
+	// defaultPluginName is the default name of the plugin.
+	defaultPluginName = "ClusterEligibility"
+)
+
+// Plugin is the scheduler plugin that performs the cluster eligibility check.
+type Plugin struct {
+	// The name of the plugin.
+	name string
+
+	// The cluster eligibility checker in use by the scheduler framework.
+	clusterEligibilityChecker *clustereligibilitychecker.ClusterEligibilityChecker
```

Comment:
nit, hmm, is it necessary?

use the handle.ClusterEligibilityChecker() directly?

Reply:
Done.